### PR TITLE
Couchpotato: init at 3.0.1 + couchpotato module

### PR DESCRIPTION
**Pull request by fadenb**
###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -1178,6 +1178,45 @@ in {
     };
   };
 
+  couchpotato = buildPythonPackage rec {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: couchpotato is used as an application and not as a library. Therefore, it doesn't belong in `python-packages.nix`. Instead, it should fit somewhere else in Nixpkgs tree based on what it is/does.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -1178,6 +1178,45 @@ in {
...
+    name = "couchpotato-${version}";
+    version = "3.0.1";
+    disabled = isPy3k;
+
+    src = pkgs.fetchurl {
+      url = "https://github.com/CouchPotato/CouchPotatoServer/archive/build/${version}.tar.gz";
+      sha256 = "1xwjis3ijh1rff32mpdsphmsclf0lkpd3phpgxkccrigq1m9r3zh";
+    };
+
+    phases = [ "unpackPhase" "patchPhase" "installPhase" "postInstall" ];
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: please remove this since there is no need for it

Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: you might want to pass `format = "other";`. That gives you a `buildPythonPackage` for non-setuptools installations.

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -1178,6 +1178,45 @@ in {
...
+
+    postPatch = ''
+      substituteInPlace CouchPotato.py --replace "dirname(os.path.abspath(__file__))" "os.path.join(dirname(os.path.abspath(__file__)), '../${python.sitePackages}')"
+    '';
+
+    installPhase = ''
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: you're making a proper package out of it. That's good. I wish upstream would do that as well.

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -1178,6 +1178,45 @@ in {
...
+      mkdir -p $out/bin/
+      mkdir -p $out/${python.sitePackages}/
+
+      cp -r libs/* $out/${python.sitePackages}/
+      cp -r couchpotato $out/${python.sitePackages}/
+
+      cp CouchPotato.py $out/bin/
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: having a binary with an extension is rather ugly. Maybe we want to rename it instead to `couchpotato` ?

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -1178,6 +1178,45 @@ in {
...
+      chmod +x $out/bin/*
+    '';
+
+    postInstall = ''
+      wrapProgram "$out/bin/CouchPotato.py" --prefix PYTHONPATH : "$PYTHONPATH:$out/${python.sitePackages}" \
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: the wrapping is done in the `fixupPhase` so this shouldn't be necessary.

File: pkgs/top-level/python-packages.nix [hunk 6]
```diff
@@ -1178,6 +1178,45 @@ in {
...
+                                            --prefix PATH : ${python}/bin
+    '';
+
+    meta = {
+      description = "Automatic movie downloading via NZBs and torrents";
+      license     = licenses.gpl3;
+      maintainers = with maintainers; [ fadenb ];
+    };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: homepage?

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by fadenb** [comment]
@FRidh: Thanks for your input. I tried to address all the issues:
* homepage: Added to metadata
* .py extension: Removed and lowercased to 'couchpotato'
* phases: Did not know about format :)
* not a library: Moved to the normal package tree
* fixup/wrapping: The automatic fixup wraps way more than just the CouchPotato.py resulting in errors as the code tries to load the shell wrapper as python. Additionally the automatic wrapper does not add ```$out/${python.sitePackages}``` to the `$PYTHONPATH` so I kept my original wrapper call but moved it to the fixupPhase.

**Comment by FRidh** [comment]
Great. But then at least use `set` instead of `prefix`.

File: pkgs/servers/couchpotato/default.nix [hunk 7]
```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchurl, pythonPackages, pkgs, lib, ... }:
```
Comment (FRidh) [inline pkgs/servers/couchpotato/default.nix]: don't pass in `pkgs`

Comment (FRidh) [inline pkgs/servers/couchpotato/default.nix]: stdenv isn't needed either, nor are the dots.

**Comment by FRidh** [comment]
> phases: Did not know about format :)

That's still undocumented actually.

**Pull request merged**